--- a/fs/namei.c
+++ b/fs/namei.c
@@ -1097,8 +1097,8 @@
 
 static int sysctl_protected_symlinks __read_mostly = 1;
 static int sysctl_protected_hardlinks __read_mostly = 1;
-int sysctl_protected_fifos __read_mostly = 1;
-int sysctl_protected_regular __read_mostly = 1;
+static int sysctl_protected_fifos __read_mostly = 2;
+static int sysctl_protected_regular __read_mostly = 2;
 
 #ifdef CONFIG_SYSCTL
 static const struct ctl_table namei_sysctls[] = {
